packages/compass-global-writes/src/store/reducer.ts

```diff
@@ -224,62 +191,71 @@ export type RootState = {
   | {
       status: ShardingStatuses.LOADING_ERROR;
       shardKey?: ShardKey;
-      shardingError?: never;
-      pollingTimeout?: never;
       loadingError: string;
+      //////////////
+      userActionInProgress?: never;
```

Comment:
this is the restriction. I had the full type in the root before, but then I removed it because it seemed unnecessary since each union part had a different definition. I can add it again for readability